scripts/analyze_csv_data.py

```diff
@@ -0,0 +1,771 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+#  This script generates all the experimental results used in the paper.
+#  It requires numpy, pandas, and matplotlib installed to run.
+#
+#   `python analyze_csv_data.py`
+#
+
+
+import os
+import numpy as np
+import pandas as pd
+import matplotlib.pyplot as plt
+import matplotlib.ticker as ticker
+from cycler import cycler
+plt.rcParams.update({'figure.max_open_warning': 0, 'font.size': 16})
+
+# This is required to generate plots that are easy to read when printed:
+plt.rcParams['axes.prop_cycle'] = cycler(
+    linestyle=[
+        '-',
+        '--',
+        ':',
+        '-.',
+        '-',
+        '--',
+        ':',
+        '-.',
+        '-',
+        '--'],
+    color=[
+        '#1f77b4',
+        '#ff7f0e',
+        '#2ca02c',
+        '#d62728',
+        '#9467bd',
+        '#8c564b',
+        '#e377c2',
+        '#7f7f7f',
+        '#bcbd22',
+        '#17becf'])
+
+
+# directory where the csv files are located
+CSV_DIR = './csv'
+
+# directory where all the figure pdf and table tex files are written to:
+OUTPUT_DIR = './results'
+FIGS_DIR = os.path.join(OUTPUT_DIR, 'figs')
+TEX_DIR = os.path.join(OUTPUT_DIR, 'tex')
+
+OPT = 'TERMINATION_REASON_OPTIMAL'
+KKT_PASSES_LIMIT = 1e5
+TIME_LIMIT_SECS = 60 * 60  # 1hr
+# shift to use for shifted geometric mean
+SGM_SHIFT = int(10)
+# penalised average runtime:
+PAR = 1.  # can be None, which removes unsolved instead of penalizing
+
+# Which scaling experiments to present
+SCALING_EXPS_TO_USE = [
+    'off,off',
+    'off,pock_chambolle alpha=1',
+    '10 rounds,off',
+    '10 rounds,pock_chambolle alpha=1',
+]
+
+# Which primal-weight experiments to present
+PRIMALWEIGHT_EXPS_TO_USE = [
+    'adaptive',
+    'Fixed 1e-0',
+]
+
+# placeholder:
+_BEST_STR = '_best_str_'
+
+# Dataset names:
+MITTELMANN_STR = 'lp_benchmark'
```

Comment:
Sigh.  We should probably change MITTELMANN_STR to LP_BENCHMARK_STR and MIPLIB_STR to MIP_RELAXATIONS_STR, for consistency with the paper and other code.

Reply:
After reconsideration, don't make this change.